csharp/src/Ice/Proxy.cs

```diff
@@ -295,27 +297,27 @@ public interface ObjectPrx : IEquatable<ObjectPrx>
     /// </summary>
     /// <returns>The router for the proxy. If no router is configured for the proxy, the return value
     /// is null.</returns>
-    RouterPrx ice_getRouter();
+    RouterPrx? ice_getRouter();
 
     /// <summary>
     /// Creates a new proxy that is identical to this proxy, except for the router.
     /// </summary>
     /// <param name="router">The router for the new proxy.</param>
     /// <returns>The new proxy with the specified router.</returns>
-    ObjectPrx ice_router(RouterPrx router);
+    ObjectPrx ice_router(RouterPrx? router);
```

Comment:
Should this API really bother to accept null?

Reply:
Yes we use it to clear the router.